ee/codegen/src/context/node-context/create-node-context.ts

```diff
@@ -35,8 +37,26 @@ export async function createNodeContext(
   switch (nodeData.type) {
     case WorkflowNodeType.SEARCH: {
       const searchNodeData = nodeData;
+
+      // Grab the associated document index from api if available to always populate document_index field
+      // with name instead of ID. We can only do this if the document index input is a constant value.
+      let documentIndex: DocumentIndexRead | null = null;
+      const inputValue = searchNodeData.inputs.find(
+        (input) => input.id === searchNodeData.data.documentIndexNodeInputId
+      )?.value;
+
+      const rule = inputValue?.rules?.[0];
+      if (rule?.type === "CONSTANT_VALUE") {
+        if (rule.data.value?.toString()) {
```

Comment:
nit: we could set a temp var for `rule.data.value?.toString()` since it's used twice here.